server/app/Icepeak/Main.hs

```diff
@@ -47,35 +46,33 @@ installHandlers core =
 
 main :: IO ()
 main = do
-  crashLogger <- SentryLogging.getCrashLogger
-  SentryLogging.runWithCrashLogger "Control loop error" crashLogger $ do
-    -- make sure output is flushed regularly
-    hSetBuffering stdout LineBuffering
+  -- make sure output is flushed regularly
+  hSetBuffering stdout LineBuffering
 
-    env <- getEnvironment
-    config <- execParser (configInfo env)
-
-    -- start logging as early as possible
-    logger <- Logger.newLogger (fromIntegral $ configQueueCapacity config)
-    loggerThread <- Async.async $ Logger.processLogRecords logger
+  env <- getEnvironment
+  config <- execParser (configInfo env)
 
+  -- start logging as early as possible
+  logger <- Logger.newLogger config
+  loggerThread <- Async.async $ Logger.processLogRecords logger
+  SentryLogging.runWithCrashLogger "Control loop error" (Logger.loggerSentryService logger) $ do
     -- setup metrics if enabled
     icepeakMetrics <- forM (configMetricsEndpoint config) $ const $ do
       void $ Prometheus.register Prometheus.Metric.GHC.ghcMetrics
       Metrics.createAndRegisterIcepeakMetrics
 
     eitherCore <- Core.newCore config logger icepeakMetrics
-    either putStrLn (runCore crashLogger) eitherCore
+    either (postLog logger LogInfo . Text.pack) runCore eitherCore
```

Comment:
This should be a `LogError`. `eitherCore` will only be `Left` when something went wrong during initialization.